public/app/features/dashboard/components/GenAI/GenAIPanelTitleButton.tsx

```diff
@@ -17,28 +18,31 @@ const TITLE_GENERATION_STANDARD_PROMPT =
   'The title should be shorter than 50 characters.';
 
 export const GenAIPanelTitleButton = ({ onGenerate, panel }: GenAIPanelTitleButtonProps) => {
-  function getMessages(): Message[] {
-    const dashboard = getDashboardSrv().getCurrent()!;
-
-    return [
-      {
-        content: TITLE_GENERATION_STANDARD_PROMPT,
-        role: Role.system,
-      },
-      {
-        content: `The panel is part of a dashboard with the title: ${dashboard.title}`,
-        role: Role.system,
-      },
-      {
-        content: `The panel is part of a dashboard with the description: ${dashboard.title}`,
-        role: Role.system,
-      },
-      {
-        content: `Use this JSON object which defines the panel: ${JSON.stringify(panel.getSaveModel())}`,
-        role: Role.user,
-      },
-    ];
-  }
-
-  return <GenAIButton messages={getMessages()} onReply={onGenerate} loadingText={'Generating title'} />;
+  const messages = React.useMemo(() => getMessages(panel), [panel]);
+  const onClick = React.useCallback(() => reportGenerateAIButtonClicked(EventSource.panelTitle), []);
+
+  return <GenAIButton messages={messages} onClick={onClick} onGenerate={onGenerate} loadingText={'Generating title'} />;
```

Comment:
Does it potentially make sense to move this reporting to the `GenAIButton` component (adding an optional prop to pass the event source / type?) If an event source / type is provided then we can log the call. IMO this probably would be easier to maintain overall

this would also remove need to send another callback prop for onClick which may be a little confusing given we already have a `onGenerate` callback prop

thoughts?

Reply:
I don't like to add properties related to tracking components. They are not business logic, and couple the implementation of the component to the tracking of the event... Not sure... 😓

Reply:
I think early on the maintainability of this approach is not a huge concern - but as we scale up these features we may want to revisit this 😬 I see your point around not wanting to contaminate core shared component logic with non business logic 😅